packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -527,7 +527,7 @@ export const useGridVirtualScroller = () => {
       height: contentHeight,
     };
 
-    if (rootProps.autoHeight && currentPage.rows.length === 0) {
+    if ((rootProps.autoHeight && currentPage.rows.length === 0) || size.height === 0) {
```

Comment:
I'm not sure we should do this.
We did this with `autoHeight`, because there was no way to achieve this with CSS.
Now, Data Grid behaves like any other CSS element in terms of its dimensions, and I think we should stick to this.
I've opened https://github.com/mui/mui-x/pull/14900 to apply the minHeight with CSS instead.